numpy/core/tests/test_dtype.py

```diff
@@ -314,6 +314,23 @@ def test_union_struct(self):
                        'formats':['i1', 'O'],
                        'offsets':[np.dtype('intp').itemsize, 0]})
 
+    @pytest.mark.parametrize(["obj", "dtype", "expected"],
+        [(3, "(3)f4,", [3, 3, 3]),
+         (np.float64(2), "(2)f4,", [2, 2]),
```

Comment:
To be clear, the first case here previously returned `[3, uninitialized, uninitialized]`, while the second case actually did the right thing.  So I considered that a simple bugfix here.